chatbot_core/utils/conversation_utils.py

```diff
@@ -20,12 +20,15 @@
 from itertools import cycle
 
 
-def create_conversation_cycle() -> cycle:
+def create_conversation_cycle(discussion_rounds: int = 1) -> cycle:
     """Cycle through conversation states"""
     from chatbot_core.utils.enum import ConversationState
 
-    return cycle([ConversationState.RESP,
-                  ConversationState.DISC,
-                  ConversationState.VOTE,
-                  ConversationState.PICK,
-                  ConversationState.IDLE])
+
+    flow = [ConversationState.RESP] + \
```

Comment:
we need to also track the actual index of the discussion round to consider injecting context from the previous discussion rounds, but for now this looks good